### PR TITLE
Fixes some issues with relation casing

### PR DESCRIPTION
**Pull request by msmith-techempower**
Also, adds the ability to specify both a table name and column names on relations.

Addresses #9

### CODE REVIEW

File: gemini/src/main/java/com/techempower/cache/CachedRelation.java [hunk 1]
```diff
@@ -192,14 +192,14 @@ protected CachedRelation(EntityStore store,
...
             + "to" + rightType.getSimpleName().toLowerCase()
         : tableName;
     this.leftColumn = (leftColumn == null)
-        ? leftType.getSimpleName().toLowerCase()
+        ? leftType.getSimpleName().toLowerCase() + "id"
         : leftColumn;
     this.rightColumn = (rightColumn == null)
-        ? rightType.getSimpleName().toLowerCase()
+        ? rightType.getSimpleName().toLowerCase() + "id"
         : rightColumn;
```
Comment (msmith-techempower) [inline gemini/src/main/java/com/techempower/cache/CachedRelation.java]: These changes mean that in order to use default annotations (`@Relation`, `@Left`, and `@Right` without specifying table or column names) the schema for a relationship table should be:
```
footobar
-------------
fooid | barid
-------------
```

File: gemini/src/main/java/com/techempower/gemini/pyxis/JsonWebToken.java [hunk 2]
```diff
@@ -97,7 +97,8 @@ public JsonWebToken(GeminiApplicationInterface application, PyxisUser user,
     this.application = application;
     this.issuedAt = issuedAt;
     bearerUserId = user.getId();
-    lastPasswordChange = user.getUserLastPasswordChange().getTime();
+    lastPasswordChange = user.getUserLastPasswordChange() == null ? 0L :
+            user.getUserLastPasswordChange().getTime();
```
Comment (msmith-techempower) [inline gemini/src/main/java/com/techempower/gemini/pyxis/JsonWebToken.java]: This fixes a bug when you have a user who has never reset his password.

File: gemini/src/main/java/com/techempower/cache/EntityStore.java [hunk 3]
```diff
@@ -1327,18 +1326,33 @@ else if (field.isAnnotationPresent(Right.class))
...
-                register(CachedRelation.of((Class<? extends Identifiable>)left.getType(), (Class<? extends Identifiable>)right.getType()).
-                    table(clazz.getSimpleName()).
-                    leftColumn(left.getName()).
-                    rightColumn(right.getName()),
-                    clazz);
+              if (getRelation((Class<EntityRelationDescriptor<Identifiable, Identifiable>>)clazz) == null) {
+                final CachedRelation.Builder<?, ?> builder = CachedRelation.of(
+                        (Class<? extends Identifiable>) left.getType(),
+                        (Class<? extends Identifiable>) right.getType());
+                if (StringHelper.isNonEmpty(annotation.table()))
```
Comment (michaelhixson) [inline gemini/src/main/java/com/techempower/cache/EntityStore.java]: The elements of an annotation can't be null, so you don't need `StringHelper` here.

```
if (!annotation.table.isEmpty())
```

File: gemini/src/main/java/com/techempower/data/EntityGroup.java [hunk 4]
```diff
@@ -110,29 +110,29 @@ public int compare(Identifiable o1, Identifiable o2)
...
+      return c1.compareTo(c2);
+    }
+  }
+
   /**
    * Gets a natural-ordering Comparator implementation for a Comparable.
    */
   @SuppressWarnings("unchecked")
   public static <T extends Comparable<? super T>> Comparator<T> naturalOrder() {
       return (Comparator<T>) EntityGroup.NaturalOrderComparator.INSTANCE;
```
Comment (michaelhixson) [inline gemini/src/main/java/com/techempower/data/EntityGroup.java]: We should get rid of this eventually and use `Comparator.naturalOrder()` instead.

File: gemini/src/main/java/com/techempower/cache/EntityStore.java (hunk 3 above)
Comment (msmith-techempower) [reply gemini/src/main/java/com/techempower/cache/EntityStore.java]: I honestly did not know this.

**Pull request merged**